hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/impl/ExecutionServiceImpl.java

```diff
@@ -395,23 +396,26 @@ public void provideDynamicMetrics(MetricDescriptor descriptor,
                 MetricDescriptor executorDescriptor = descriptor
                         .copy()
                         .withPrefix("internal-executor")
-                        .withDiscriminator("executor", executorService.getName());
+                        .withDiscriminator("executor", executorService.getName())
+                        .withExcludedTarget(MANAGEMENT_CENTER);
                 context.collect(executorDescriptor, executorService);
             }
 
             for (ManagedExecutorService executorService : durableExecutors.values()) {
                 MetricDescriptor executorDescriptor = descriptor
                         .copy()
                         .withPrefix("durable-executor")
-                        .withDiscriminator("executor", executorService.getName());
+                        .withDiscriminator("executor", executorService.getName())
+                        .withExcludedTarget(MANAGEMENT_CENTER);
```

Comment:
I think we should keep metrics for durable and scheduled durable executors available for MC, as we may decide to implement support for these user defined executors in MC in future.

Reply:
I see your point. Note that we provide executor service metrics from two places:
1) `ExecutionServiceImpl`: it provides metrics about the pool's size backing the executor and about its queue metrics. See `ManagedExecutorService` for the concrete probes. This class provides metrics for all types of executors. Internals and user-created ones. Also, it includes durable and scheduled executors as well.
2) `DistributedExecutorService`: This provides `pending`, `started` etc metrics (see `LocalExecutorStatsImpl` for the probes) for the distributed executors, but lacks providing the same for the durable and for the scheduled executors.

MC currently relies on the set of the metrics provided by 2), and is not interested in the backing executor metrics provided by 1). The point is, even if we remove the exclusion for the durable and the scheduled executor metrics in 1), MC will not get the metrics it would need for the durable and scheduled executors. 

I think what we need is providing the same set of metrics for the durable and the scheduled executors in 2), which currently doesn't even exist.

Additionally, we need to unify the metrics to make it easier to correlate if we make MC interested in the 1) set of metrics. This is needed, since 1) provides like this:
```
[executor=myExecutor,unit=count,metric=internal-executor.remainingQueueCapacity]
[executor=myExecutor,unit=count,metric=internal-executor.maximumPoolSize]
[executor=myExecutor,unit=count,metric=internal-executor.poolSize]
[executor=myExecutor,unit=count,metric=internal-executor.completedTaskCount]
[executor=myExecutor,unit=count,metric=internal-executor.queueSize]

[executor=myDurableExecutor,unit=count,metric=durable-executor.remainingQueueCapacity]
[...]

[executor=myScheduledExecutor,unit=count,metric=scheduled-executor.remainingQueueCapacity]
[...]
```
while 2) like this:
```
[name=myExecutor,unit=count,metric=executor.pending]
[name=myExecutor,unit=count,metric=executor.started]
[name=myExecutor,unit=count,metric=executor.completed]
[name=myExecutor,unit=count,metric=executor.cancelled]
[name=myExecutor,unit=count,metric=executor.totalStartLatency]
[name=myExecutor,unit=count,metric=executor.totalExecutionTime]
```

So I'd change 1): `executor=` -> `name=` and would include `internal-` prefix for durable and scheduled executors as well to have this:
```
[name=myExecutor,unit=count,metric=internal-executor.remainingQueueCapacity]
[...]

[name=myDurableExecutor,unit=count,metric=internal-durable-executor.remainingQueueCapacity]
[...]

[name=myScheduledExecutor,unit=count,metric=internal-scheduled-executor.remainingQueueCapacity]
[...]
```

Reply:
> I think what we need is providing the same set of metrics for the durable and the scheduled executors in 2), which currently doesn't even exist.

Makes sense. I've overlooked absence of these metrics. Then I'm fine with excluding them, as they're effectively `internal-` ones.

> So I'd change 1): executor= -> name= and would include internal- prefix for durable and scheduled executors as well to have this:

Sounds like a logical change to me.